lib/rules/email.js

```diff
@@ -1,18 +1,17 @@
 "use strict";
 
-const PRECISE_PATTERN = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+/* eslint no-useless-escape: off */
+const PRECISE_PATTERN = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
 const BASIC_PATTERN = /^\S+@\S+\.\S+$/;
 
 module.exports = function checkEmail(value, schema) {
 	if (typeof value !== "string") {
 		return this.makeError("string");
 	}
 
-	let pattern;
-	if (schema.mode == "precise")
+	let pattern = BASIC_PATTERN;
```

Comment:
Why we should create additional variable here instead of using `PRECISE_PATTERN` constant?